### ISSUE TITLE
css file not getting included in epub

### ISSUE THREAD

**Issue by Shreeshrii** [jgm/pandoc#3332]
I am trying to include a devanagari font in an epub created with xelatex source. While the font is getting embedded in the epub, it seems to be using only the default stylesheet.css in the epub.
```

pandoc -t epub test1.tex -o test1-lohitwoff.epub --latex-engine=xelatex --epub-stylesheet=lohitwoff.css --epub-embed-font=Lohit-Devanagari.woff

 unzip -c test1-lohitwoff.epub content.opf
Archive:  test1-lohitwoff.epub
  inflating: content.opf
<?xml version="1.0" encoding="UTF-8"?>
<package version="2.0" xmlns="http://www.idpf.org/2007/opf" unique-identifier="epub-id-1">
  <metadata xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:opf="http://www.idpf.org/2007/opf">
    <dc:identifier id="epub-id-1">urn:uuid:33f5310f-6165-4244-8004-13b4bd5ef683</dc:identifier>
    <dc:date>2016-12-29T16:11:42Z</dc:date>
    <dc:language>en-US</dc:language>
  </metadata>
  <manifest>
    <item id="ncx" href="toc.ncx" media-type="application/x-dtbncx+xml" />
    <item id="style" href="stylesheet.css" media-type="text/css" />
    <item id="nav" href="nav.xhtml" media-type="application/xhtml+xml" />
    <item id="title_page" href="title_page.xhtml" media-type="application/xhtml+xml" />
    <item id="ch001" href="ch001.xhtml" media-type="application/xhtml+xml" />
    <item id="Lohit-Devanagari" href="Lohit-Devanagari.woff" media-type="application/x-font-woff" />
  </manifest>
  <spine toc="ncx">
    <itemref idref="title_page" linear="no" />
    <itemref idref="nav" linear="no" />
    <itemref idref="ch001" />
  </spine>
  <guide>
    <reference type="toc" title="UNTITLED" href="nav.xhtml" />
  </guide>
</package>
```

**Comment by jgm** [comment]
Pandoc uses the *name* `stylesheet.css` even if you specify a custom stylesheet.  Look at the contents of `stylesheet.css` in the epub container; it should be the same as your `lohitwoff.css`.

**Comment by Shreeshrii** [comment]
It seems my emailed reply with attachments did not come through. I will post the info via GitHub.

**Comment by Shreeshrii** [comment]
unzip test1-lohitwoff.epub
Archive:  test1-lohitwoff.epub
 extracting: mimetype
  inflating: META-INF/container.xml
  inflating: META-INF/com.apple.ibooks.display-options.xml
  inflating: stylesheet.css
  inflating: title_page.xhtml
  inflating: content.opf
  inflating: toc.ncx
  inflating: nav.xhtml
  inflating: ch001.xhtml
  inflating: Lohit-Devanagari.woff

$ unzip -c test1-lohitwoff.epub stylesheet.css
Archive:  test1-lohitwoff.epub
  inflating: stylesheet.css
/* This defines styles and classes used in the book */
body { margin: 5%; text-align: justify; font-size: medium; }
code { font-family: monospace; }
h1 { text-align: left; }
h2 { text-align: left; }
h3 { text-align: left; }
h4 { text-align: left; }
h5 { text-align: left; }
h6 { text-align: left; }
h1.title { }
h2.author { }
h3.date { }
ol.toc { padding: 0; margin-left: 1em; }
ol.toc li { list-style-type: none; margin: 0; padding: 0; }

It is not including the custom css info from the file but seems to have the default css included.

**Comment by Shreeshrii** [comment]
I would appreciate if you can suggest the best command for making epub from xetex files similar to attached which have most of the text in Sanskrit language, Devanagari script.

I was trying to attach the epub, tex source and css but GitHub does not allow the upload. I will make a zip file and upload.

**Comment by mb21** [comment]
So what you're saying is that the `stylesheet.css` in the generated ePub does not have the contents of your `lohitwoff.css` even though you're using the `--epub-stylesheet=lohitwoff.css` option, right?

Are you using the newest pandoc version? Also, can you post your `lohitwoff.css` ?

**Comment by Shreeshrii** [comment]
[pandoc.zip](https://github.com/jgm/pandoc/files/687124/pandoc.zip)

pandoc -v
pandoc 1.12.2.1
Compiled with texmath 0.6.5.2, highlighting-kate 0.5.5.1.

on
Linux sanskrit 4.4.0-57-generic #78~14.04.1-Ubuntu SMP Sat Dec 10 00:14:47 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

**Comment by jgm** [comment]
That's an extremely old version of pandoc.  We're on 1.19.1
now -- and we supply a linux 64-bit binary package that
should work on your system.  Can you try with that and see
if you still have the problem?

+++ Shreeshrii [Jan 05 17 03:32 ]:

**Comment by Shreeshrii** [comment]
Thanks! I will request the administrator of the machine to install the new version.

Can the new version be used under bash on Windows 10?

**Comment by ickc** [comment]
@Shreeshrii,

regarding WSL, you can try to install the deb which is complied and see what happened. I tried installing on WSL through cabal (built fom source) with no success. It is too new and too many things are supported right now. (they rightly said it is beta, and is included in stable Windows 10 because there's so many requests for it despite its status. Their main goal is "web-development related tools" first, so Haskell shouldn't be in their priority list. But you can give the deb a try and let us know if it works).

**Comment by ickc** [comment]
@Shreeshrii 

>Thanks! I will request the administrator of the machine to install the new version.

I think there should be a way to unpack it without sudo. A quick search give me [this from askubuntu](http://askubuntu.com/questions/339/how-can-i-install-a-package-without-root-access). So you probably can bypass the admin and use the pandoc & pandoc-citeproc executables with your own PATH. This way you can also always use the latest available version without waiting them to update (depending on the admin, they might just not do it for you, or they might install the latest available through apt-get in ubuntu 14.04, which is another outdated pandoc.)

**Comment by Shreeshrii** [comment]
I installed the latest msi version under windows 10. I am getting the following error in it, 
```

Microsoft Windows [Version 10.0.14393]
(c) 2016 Microsoft Corporation. All rights reserved.

C:\Users\User\shree\tex>pandoc -t epub test1.tex -o test1-lohitwoff.epub --latex-engine=xelatex --epub-stylesheet=lohitwoff.css --epub-embed-font=Lohit-Devanagari.woff
pandoc:
Error at "input" (line 324, column 82):
unexpected '}'
\end{center}
                                                                                 ^
CallStack (from HasCallStack):
  error, called at src\Text\Pandoc\Error.hs:66:13 in pandoc-1.19.1-BhTo5eH5ZEaHCcu9Z0Nwfe:Text.Pandoc.Error

C:\Users\User\shree\tex>
```

**Comment by jgm** [comment]
INSTALL.md contains instructions for extracting the binary
from the deb container without sudo.

**Comment by jgm** [comment]
This means there is something in your tex file that pandoc
can't understand.  You can look at that position to see what
it is.  If it looks like a bug in pandoc you can report
that, but try to understand what pandoc is stumbling on
first.

+++ Shreeshrii [Jan 05 17 20:06 ]:

**Comment by Shreeshrii** [comment]
The same tex file worked with the older version of pandoc and created an
epub and could be read by the epub readers.

Similar xelatex sources are regularly used for creating PDFs for Sanskrit
documents.org.

Do you want this to be reported as a separate bug?

I have already provided a zip file with all relevant input files.

- excuse the brevity, sent from mobile

**Comment by jgm** [comment]
+++ Shreeshrii [Jan 06 17 05:35 ]:
>   The same tex file worked with the older version of pandoc and created
>   an
>   epub and could be read by the epub readers.

Yes, some earlier versions were more forgiving and just
skipped ahead when they encountered something they didn't
understand.

>   Similar xelatex sources are regularly used for creating PDFs for
>   Sanskrit
>   documents.org.

Yes, this doesn't indicate an error in tex, just something
in tex that pandoc doesn't know how to handle.

>   Do you want this to be reported as a separate bug?

Yes, it should be a separate issue, but please, if you make
a bug report, try to create a minimal test case rather than
a gigantic file.

**Comment by ickc** [comment]
> @Shreeshrii
> 
> Thanks! I will request the administrator of the machine to install the new version.
> I think there should be a way to unpack it without sudo. A quick search give me this from askubuntu. So you probably can bypass the admin and use the pandoc & pandoc-citeproc executables with your own PATH. This way you can also always use the latest available version without waiting them to update (depending on the admin, they might just not do it for you, or they might install the latest available through apt-get in ubuntu 14.04, which is another outdated pandoc.)

After communicated with @jgm, he said the install method without the need of sudo is already given in [Pandoc - Installing](http://pandoc.org/installing.html#linux): look at the 3rd bullet under "Linux".

**Comment by Shreeshrii** [comment]
Thank you all for your prompt and helpful responses. I was able to install
the latest Deb version of pandoc in my user directory.

I also deleted most if the preamble from the tex file, and was able to
generate the epub without any error.

The css is now getting included in the epub, so this issue can be closed.

The css, in terms of Colors for the headings is being applied, I need to
check regarding the font. I shall open a new issue with MWE if needed.

Appreciate the help.

- excuse the brevity, sent from mobile
